cmd/plugin/secret_generation.go

```diff
@@ -0,0 +1,475 @@
+package main
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const kubectl = "kubectl"
+
+var rootCmd = &cobra.Command{
+	Use:   "secret-generation",
+	RunE:  secretGeneration,
+	Short: "Create a kubeconfig secret for the pod lister controller",
+}
```

Comment:
I think the name and description probably needs updated. "secret-generation" is a bit vague, its specifically for creating a kubeoncifg secret for multi cluster communication and probably should reflect that. The description, fair enough was copied as is from the hack script i added, but is actually wrong. Maybe just "Create a kubeconfig secret" would be better.